beetmoverscript/src/beetmoverscript/task.py

```diff
@@ -34,9 +34,39 @@ def validate_task_schema(context):
     client.validate_task_schema(context, schema_key=schema_key)
 
 
+def _get_bucket_prefixes(script_config):
+    return _get_scope_prefixes(script_config, "bucket")
+
+
+def _get_action_prefixes(script_config):
+    return _get_scope_prefixes(script_config, "action")
+
+
+def _get_scope_prefixes(script_config, sub_namespace):
+    prefixes = script_config["taskcluster_scope_prefixes"]
+    prefixes = [prefix if prefix.endswith(":") else "{}:".format(prefix) for prefix in prefixes]
+    return ["{}{}:".format(prefix, sub_namespace) for prefix in prefixes]
+
+
+def _extract_scopes_from_unique_prefix(scopes, prefixes):
+    scopes = [scope for scope in scopes for prefix in prefixes if scope.startswith(prefix)]
+    _check_scopes_exist_and_all_have_the_same_prefix(scopes, prefixes)
+    return scopes
+
+
+def _check_scopes_exist_and_all_have_the_same_prefix(scopes, prefixes):
+    for prefix in prefixes:
+        if all(scope.startswith(prefix) for scope in scopes):
+            break
+    else:
+        raise ScriptWorkerTaskException("Scopes must exist and all have the same prefix. " "Given scopes: {}. Allowed prefixes: {}".format(scopes, prefixes))
```

Comment:
nit: Any reason not to use f-strings here? I also tend to like to format lists per-line in error messages, e.g:
```
scope_str = " -".join(scopes)
prefix_str = " -".join(prefixes)
... f"Given scopes:\n  -{scope_str}\n\nAllowed prefixes:\n -{prefix_str}"...
```

though it seems like this probably won't ever happen so, meh. Up to you.